tests/foreman/maintain/test_backup_restore.py

```diff
@@ -409,6 +408,88 @@ def test_negative_restore_baddir(sat_maintain, setup_backup_tests):
     assert BADDIR_MSG in str(result.stdout)
 
 
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+def test_positive_puppet_backup_restore(
+    sat_maintain,
+    setup_backup_tests,
+    module_synced_repos,
+    backup_type,
+):
+    """Puppet backup/restore test.
+
+    :id: 0f8555dc-8c1f-47cd-b5f6-5655d98564cd
+
+    :parametrized: yes
+
+    :steps:
+        1. Enable puppet on the satellite
+        2. create a backup of different types
+        3. check that appropriate files are created
+        4. restore the backup (installer --reset-data is run in this step)
+        5. check system health
+        6. check the content was restored
+
+    :expectedresults:
+        1. backup succeeds
+        2. expected files are present in the backup
+        3. restore succeeds
+        4. system health check succeeds
+        5. content is present after restore
+
+    :customerscenario: true
+
+    :BZ: 2158896
+    """
+    # Enable puppet on the Satellite
+    sat_maintain.enable_puppet_satellite()
+
+    subdir = f'{BACKUP_DIR}backup-{gen_string("alpha")}'
+    instance = 'satellite' if type(sat_maintain) is Satellite else 'capsule'
+    result = sat_maintain.cli.Backup.run_backup(
+        backup_dir=subdir,
+        backup_type=backup_type,
+        options={'assumeyes': True, 'plaintext': True, 'skip-pulp-content': False},
+    )
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+
+    # Check for expected files
+    backup_dir = re.findall(fr'{subdir}\/{instance}-backup-.*-[0-5][0-9]', result.stdout)[0]
+    files = sat_maintain.execute(f'ls -a {backup_dir}').stdout.split('\n')
+    files = [i for i in files if not re.compile(r'^\.*$').search(i)]
```

Comment:
also non-blocking but in the future, you may want to consider using `filter` here.